scripts/jenkins/ardana/gerrit/build_test_package.py

```diff
@@ -0,0 +1,135 @@
+        fetch_obj = self._change_object['revisions'][current_revision]['fetch']
+        self.url = fetch_obj['anonymous http']['url']
+        self.ref = fetch_obj['anonymous http']['ref']
+        self.target = self._change_object['branch']
+
+    def prep_workspace(self):
+        if os.path.exists('./source'):
+            shutil.rmtree('./source')
+        os.mkdir('source')
+        with cd('source'):
+            sh.git('clone', self.url, '%s.git' % self.project)
+            with cd('%s.git' % self.project):
+                sh.git('checkout', '-b', 'test-merge', self.target)
+                sh.git('fetch', self.url, self.ref)
+                sh.git('merge', '--no-edit', 'FETCH_HEAD')
+
+
+def create_test_project(change_id, develproject):
+    testproject = '%s:Testing-%s' % (develproject, change_id)
+    repo_metadata = """
+<project name="%(testproject)s">
+  <title>Autogenerated CI project</title>
+  <description/>
+  <link project="%(develproject)s"/>
+  <person userid="opensuseapibmw" role="maintainer"/>
+  <publish>
+    <enable repository="standard"/>
+  </publish>
+  <repository name="standard" rebuild="direct" block="local"
+      linkedbuild="localdep">
+    <path project="%(develproject)s" repository="SLE_12_SP3"/>
+    <arch>x86_64</arch>
+  </repository>
+</project>
+""" % {'testproject': testproject, 'develproject': develproject}
+
+    with tempfile.NamedTemporaryFile() as meta:
+        meta.write(repo_metadata)
+        meta.flush()
+        sh.osc('-A', 'https://api.suse.de', 'api', '-T', meta.name,
+            '/source/%s/_meta' % testproject)
+
+    return testproject
+
+
+def create_test_package(change, testproject):
+    package_name = project_map()[change.project]
+    sh.osc('-A', 'https://api.suse.de', 'copypac', '--keep-link',
+        'Devel:Cloud:8', package_name, testproject)
+    sh.osc('-A', 'https://api.suse.de', 'checkout', testproject, package_name)
+    source_dir = '%s/source/%s.git' % (os.getcwd(), change.project)
+    with cd('%s/%s' % (testproject, package_name)):
+        with open('_service', 'r+') as service_file:
+            service_def = service_file.read()
+            service_def = re.sub(r'<param name="url">.*</param>',
+                                 '<param name="url">%s</param>' % source_dir, service_def)
+            service_def = re.sub(r'<param name="revision">.*</param>',
+                                 '<param name="revision">test-merge</param>', service_def)
+            service_file.seek(0)
+            service_file.write(service_def)
+            service_file.truncate()
+        sh.osc('rm', glob.glob('%s*.obscpio' % package_name))
+        sh.osc('service', 'disabledrun')
+        sh.osc('add', glob.glob('%s*.obscpio' % package_name))
+        sh.osc('commit', '-m', 'Testing change %s' % change.id)
+        # Wait for build to be scheduled
+        while 'unknown' in sh.osc('results'):
+            sleep(3)
```

Comment:
This should be `time.sleep(3)`

and flake8 (via travis-ci) has some other complaints about this file that will need to be addressed.